### PR TITLE
Simplified code & added support for universal apps

### LINKED ISSUE #5: Not compatible with universal
This implementation does not work with angular universal because the _window function does not check if window is set.

Please look at http://brianflove.com/2018/01/11/angular-window-provider/

### PR DESCRIPTION
**Pull request by fluidsonic**
Added support for Angular Universal. Fixes #5.

### CODE REVIEW

**Comment by maxisam** [comment]
Thank you for your help. However, I wonder if we should use `providedIn:'root' `, so we can avoid issues when this is being loaded in a lazyload module.

Check this out https://github.com/angular/angular/issues/23715

**Comment by fluidsonic** [comment]
I'm fine with putting it into the hierarchy root but I have no idea whether there are any side-effects on more complex projects.

And putting it into root would basically make `NgxWindowTokenModule` obsolete, wouldn't it?

Let me know if you want to move on with root and I'll update my commit :)

**Comment by maxisam** [comment]
@fluidsonic it seems to me that putting it into root would avoid issues. There are couple issues reported related to lazy load module already.

It seems like this change would make `NgxWindowTokenModule` obsolete but I honestly don't know how it will work without module. We probably need to test it. I will make a beta version after you change it to root and remove the module from public api.

Thanks!

**Comment by fluidsonic** [comment]
@maxisam I've updated all relevant files. What do you think?

File: README.md [hunk 1]
```diff
@@ -11,6 +11,10 @@ This project was generated with [Angular CLI](https://github.com/angular/angular
 
 It is the same way how angular handle document. With this module, you can use WINDOW token just like DOCUMENT token.
 
+# Breaking Change after version 1.0.2
```
Comment (maxisam) [inline README.md:14]: do you mean 2.0.0?

Comment (fluidsonic) [reply README.md:14]: The convention in the README seems to be to mention the **previous** version, hence "**after** version 1.0.2".

Comment (maxisam) [reply README.md:14]: I see. Thanks!

**Comment by pburkindine** [comment]
Thanks guys!

**Comment by maxisam** [comment]
Guys, please test 2.0.0-beta.01. I haven't test it in lazyloading case but other than that it seems working fine.

**Pull request merged**